examples/api/lib/widgets/nested_scroll_view/nested_scroll_view.3.dart

```diff
@@ -0,0 +1,197 @@
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: NestedScrollView(
+        headerSliverBuilder: (BuildContext context, bool innerBoxIsScrolled) {
+          // These are the slivers that show up in the "outer" scroll view.
+          return <Widget>[
+            SliverOverlapAbsorber(
+              // This widget takes the overlapping behavior of the SliverAppBar,
+              // and redirects it to the SliverOverlapInjector below. If it is
+              // missing, then it is possible for the nested "inner" scroll view
+              // below to end up under the SliverAppBar even when the inner
+              // scroll view thinks it has not been scrolled.
+              // This is not necessary if the "headerSliverBuilder" only builds
+              // widgets that do not overlap the next sliver.
+              handle: NestedScrollView.sliverOverlapAbsorberHandleFor(context),
+              sliver: SliverAppBar(
+                title: const Text('Books'), // This is the title in the app bar.
+                pinned: true,
+                expandedHeight: 150.0,
+                // The "forceElevated" property causes the SliverAppBar to show
+                // a shadow. The "innerBoxIsScrolled" parameter is true when the
+                // inner scroll view is scrolled beyond its "zero" point, i.e.
+                // when it appears to be scrolled below the SliverAppBar.
+                // Without this, there are cases where the shadow would appear
+                // or not appear inappropriately, because the SliverAppBar is
+                // not actually aware of the precise position of the inner
+                // scroll views.
+                forceElevated: innerBoxIsScrolled,
+                bottom: TabBar(
+                  controller: _tabController,
+                  // These are the widgets to put in each tab in the tab bar.
+                  tabs: <Widget>[
+                    for (final String name in _tabs) Tab(text: name),
+                  ],
+                ),
+              ),
+            ),
+          ];
+        },
+        body: TabBarView(
+          controller: _tabController,
+          // These are the contents of the tab views, below the tabs.
+          children: <Widget>[
+            for (final (int index, String name) in _tabs.indexed)
+              TabViewContentExample(
+                name: name,
+                isSelected: index == _selectedIndex,
+              ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class TabViewContentExample extends StatefulWidget {
+  const TabViewContentExample({
+    required this.name,
+    required this.isSelected,
+    super.key,
+  });
+
+  final String name;
+  final bool isSelected;
+
+  @override
+  State<TabViewContentExample> createState() => TabViewContentExampleState();
+}
+
+class TabViewContentExampleState extends State<TabViewContentExample>
```

Comment:
Might as well follow the "public widget, private state" convention:

```suggestion
class _TabViewContentExampleState extends State<TabViewContentExample>
```

Reply:
Thanks! I fixed it!